llumnix/arg_utils.py

```diff
@@ -22,6 +22,77 @@
 from llumnix.config import LlumnixConfig, get_llumnix_config
 from llumnix.config.default import _C
 
+
+class LlumnixArgumentParser(argparse.ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        self.cur_namespace = "llumnix"
+        super().__init__(*args, **kwargs)
+
+    def set_namespace(self, namespace: str):
+        self.cur_namespace = namespace
+
+    def add_argument(self, *args, **kwargs):
+        if self.cur_namespace == 'llumnix' and "--help" not in args:
+            assert 'default' not in kwargs or kwargs['default'] is None, \
+                f"Do not set the default value for '{args[0]}' in CLI, or set default value to None. " \
+                f"The default value will be retrieved from config/default.py in get_llumnix_config."
+            if kwargs.get('action') == 'store_true':
+                kwargs['default'] = None
+        super().add_argument(*args, **kwargs)
+
+
+# All the default values of llumnix arguments are setted in default.py. So all the arguments here are setted to None.
+
+@dataclass
+class LlumnixEntrypointsArgs:
+    launch_ray_cluster: bool = None
+    ray_cluster_port: int = None
+    queue_type: str = None
+    request_output_queue_port: int = None
+    disable_log_requests_server: bool = None
+    log_request_timestamps: bool = None
+    config_file: bool = None
+
+    def __post_init__(self):
+        for attr in dataclasses.fields(self):
+            if getattr(self, attr.name) is None:
+                setattr(self, attr.name, getattr(_C.SERVER, attr.name.upper()))
```

Comment:
add a  checker for the default value must be None

Reply:
Get